src/app/modules/main/shared_urls/module.nim

```diff
@@ -53,8 +53,12 @@ method parseSharedUrl*(self: Module, url: string): UrlDataDto =
   return self.controller.parseSharedUrl(url)
 
 method parseCommunitySharedUrl*(self: Module, url: string): string =
-  let communityData = self.controller.parseCommunitySharedUrl(url)
-  return $communityData
+  let parsedUrl = self.controller.parseSharedUrl(url)
+  let shard = parsedUrl.shard
+  var communityJson = parsedUrl.community.toJsonNode()
+  communityJson["shardCluster"] = %shard.cluster
+  communityJson["shardIndex"] = %shard.index
+  return $communityJson
```

Comment:
Yeah, I agree that it would be better if it was part of the original object in status-go. So yeah, I had to hack this up to get the shard info to the QML from the URL. I created this issue to track it in status-go: https://github.com/status-im/status-go/issues/4318

I'll try to merge it in Nim for now